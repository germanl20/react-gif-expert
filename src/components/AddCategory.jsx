import { useState } from "react"

export const AddCategory = ({onNewCategory}) => {
    const [inputValue, setInputValue] = useState('');

    const onInputChange = ({target}) => {
        setInputValue(target.value);
    }

    const onSubmit = (event) => {
        event.preventDefault();
        const newCategory = inputValue.trim();

        if(newCategory.length < 1) return;

        //onAddCategory(categories => [inputValue, ...categories]);
        onNewCategory(newCategory);
        setInputValue('');
    }

    return (
        <form onSubmit={onSubmit}>
            <input
                type="text"
                placeholder="Buscar Gifs"
                value={inputValue}
                onChange={onInputChange}
            />
        </form>
    )
}
