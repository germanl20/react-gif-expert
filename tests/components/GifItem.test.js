import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

const title = 'Titulo';
const url = 'https://prueba.com/';

describe('Pruebas en GifItem', () => {
    test('Evaluar Snapshot', () => {
        const { container } = render(<GifItem title={ title } url={ url } />);
        expect(container).toMatchSnapshot();
    })

    test('Debe existir el titulo y la url en la img', () => {
        render(<GifItem title={ title } url={ url } />);
        const { src, alt } = screen.getByRole('img');

        expect(src).toBe(url);
        expect(alt).toBe(title);
    })

    test('Debe existir el titulo', () => {
        render(<GifItem title={ title } url={ url } />);
        expect(screen.getByText(title)).toBeTruthy();
    })
})