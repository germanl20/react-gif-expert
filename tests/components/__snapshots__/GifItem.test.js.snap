// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem Evaluar Snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Titulo"
      src="https://prueba.com/"
    />
    <p>
      Titulo
    </p>
  </div>
</div>
`;
